common/lib/capa/capa/capa_problem.py

```diff
@@ -875,10 +874,65 @@ def _preprocess_problem(self, tree):  # private
                 entry.attrib['id'] = "%s_%i_%i" % (self.problem_id, response_id, answer_id)
                 answer_id = answer_id + 1
 
-            question_id = u'{}_{}'.format(self.problem_id, response_id)
-            label = ''
-            element_to_be_deleted = None
+            self.response_a11y_data(response, inputfields, responsetype_id, problem_data)
 
+            # instantiate capa Response
+            responsetype_cls = responsetypes.registry.get_class_for_tag(response.tag)
+            responder = responsetype_cls(response, inputfields, self.context, self.capa_system, self.capa_module)
+            # save in list in self
+            self.responders[response] = responder
+
+        # get responder answers (do this only once, since there may be a performance cost,
+        # eg with externalresponse)
+        self.responder_answers = {}
+        for response in self.responders.keys():
+            try:
+                self.responder_answers[response] = self.responders[response].get_answers()
+            except:
+                log.debug('responder %s failed to properly return get_answers()',
+                          self.responders[response])  # FIXME
+                raise
+
+        # <solution>...</solution> may not be associated with any specific response; give
+        # IDs for those separately
+        # TODO: We should make the namespaces consistent and unique (e.g. %s_problem_%i).
+        solution_id = 1
+        for solution in tree.findall('.//solution'):
+            solution.attrib['id'] = "%s_solution_%i" % (self.problem_id, solution_id)
+            solution_id += 1
+
+        return problem_data
+
+    def response_a11y_data(self, response, inputfields, responsetype_id, problem_data):
+        """
+        Construct data to be used for a11y.
+
+        Arguments:
+            response (object): xml response object
+            inputfields (list): list of inputfields in a responsetype
+            responsetype_id (str): responsetype id
+            problem_data (dict): dict to be filled with response data
+        """
+        element_to_be_deleted = None
+        label = ''
+
+        if len(inputfields) > 1:
```

Comment:
@muhammad-ammar Can we make sure that if input-fields are empty we don't execute this block ?